components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenBindingMgtDAO.java

```diff
@@ -76,4 +76,14 @@ default Optional<TokenBinding> getTokenBindingByBindingRef(String tokenId, Strin
      * @throws IdentityOAuth2Exception in case of failure.
      */
     void deleteTokenBinding(String tokenId) throws IdentityOAuth2Exception;
+
+    /**
+     * Obtain token binding from refresh token.
+     *
+     * @param refreshToken Refresh token sent in the request.
+     * @param isTokenHashingEnabled Whether token hashing is enabled.
+     * @throws IdentityOAuth2Exception An exception is thrown if an error occurs while obtaining the token binding.
+     */
+    TokenBinding getBindingFromRefreshToken(String refreshToken, boolean isTokenHashingEnabled)
```

Comment:
Make this a default method

Reply:
If this can return a null lets make it return a optional as well.

https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2240/files#diff-273acee4a07b082e208caffd96314389f2242a9d7625dce4e676a58c3966f172R48

Reply:
Fixed in 3da41018b6a5b63ad5435d77db06c498340e3af0